### PR TITLE
Add wait methods for dns stuff

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Add wait methods for dns stuff

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py [hunk 1]
```diff
@@ -29,12 +29,13 @@ def setUp(self):
...
+        if self.zone:
+            try:
                 self.client.delete_zone(self.zone)
-        except openstack.exceptions.SDKException as e:
-            _logger.warning('Got exception during clearing resources %s'
-                            % e.message)
+            except openstack.exceptions.SDKException as e:
+                _logger.warning('Got exception during clearing resources %s'
+                                % e.message)
+            self.client.wait_for_delete_zone(self.zone)
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: I would rather put wait_for_delete also under the try, cause here we can get timeout

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py [hunk 2]
```diff
@@ -47,13 +48,15 @@ def _create_zone(self, zone_name=None, router_id=None, zone_type='public'):
                 )
             except openstack.exceptions.BadRequestException:
                 self.zone = self.client.find_zone(zone_name)
+            self.client.wait_for_zone(self.zone)
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: also here - get it under the try

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py [hunk 3]
```diff
@@ -47,13 +48,15 @@ def _create_zone(self, zone_name=None, router_id=None, zone_type='public'):
...
             return
         try:
             self.zone = self.client.create_zone(
                 name=zone_name
             )
         except openstack.exceptions.BadRequestException:
             self.zone = self.client.find_zone(zone_name)
+        self.client.wait_for_zone(self.zone)
```
Comment (gtema) [inline otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: and what if you do wait_for_zone just once? you have zone from one or another create. You can do `wait` only once afterwards using received zone

**Review by gtema — CHANGES_REQUESTED** [review]

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py (hunk 1 above)
Comment (Polina-Gubina) [reply otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: fixed

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py (hunk 2 above)
Comment (Polina-Gubina) [reply otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: fixed

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py (hunk 3 above)
Comment (Polina-Gubina) [reply otcextensions/tests/functional/sdk/dns/v2/test_zone.py]: fixed

**Review by gtema — DISMISSED** [review]
please also add functions upstream

**Review by gtema — APPROVED** [review]

**Pull request merged**